tests/e2e/project.spec.ts

```diff
@@ -337,3 +337,58 @@ test("only one modal can be open at a time", async ({ page }) => {
   await expect(page.locator("text=bob hyy1k6g")).not.toBeVisible();
   await expect(page.locator("text=feature/branch")).not.toBeVisible();
 });
+
+test.describe("browser error handling", () => {
+  test("error appears when folder can't be loaded", async ({ page }) => {
+    await page.route(
+      "**/v1/projects/rad:git:hnrkgd7sjt79k4j59ddh11ooxg18rk7soej8o/tree/530aabdcc80397af254bc488b767169b92496e81/markdown/",
+      route => route.fulfill({ status: 500 }),
+    );
+
+    await page.goto(sourceBrowsingFixture);
+    await page.locator("text=markdown/").click();
+
+    const error = page.locator("text=Not able to expand directory");
+    await expect(error).toBeVisible();
+  });
+  test("error appears when file can't be loaded", async ({ page }) => {
+    await page.route(
+      "**/v1/projects/rad:git:hnrkgd7sjt79k4j59ddh11ooxg18rk7soej8o/blob/530aabdcc80397af254bc488b767169b92496e81/.hidden?highlight=true&theme=base16-ocean.dark",
+      route => route.fulfill({ status: 500 }),
+    );
+
+    await page.goto(sourceBrowsingFixture);
+    await page.locator("text=.hidden").click();
+
+    const error = page.locator("text=Not able to load file");
+    await expect(error).toBeVisible();
+  });
+  test("error appears when navigating to missing file", async ({ page }) => {
+    await page.route(
+      "**/v1/projects/rad:git:hnrkgd7sjt79k4j59ddh11ooxg18rk7soej8o/blob/530aabdcc80397af254bc488b767169b92496e81/.hidden?highlight=true&theme=base16-ocean.dark",
+      route => route.fulfill({ status: 500 }),
+    );
+
+    await page.goto(`${sourceBrowsingFixture}/tree/master/.hidden`);
+
+    const error = page.locator("text=Not able to load file");
+    await expect(error).toBeVisible();
+  });
+  test("error appears when navigating to nested folder that can't be loaded", async ({
```

Comment:
And this test seems to also test the same thing as `error appears when folder can't be loaded`, no? Or are we hitting different code paths with it?

Reply:
Yeah I removed it